### PR TITLE
Port qmstr to Cobra.

### LINKED ISSUE #286: Port qmstr to use Cobra
Standardize the command line interfaces. Use a similar command/subcommand hierarchy. Generate the command line reference. Create a clear separation between `qmstrctl` (a tool to interact with the master) and `qmstr` (a tool to perform client-side operations.

### PR DESCRIPTION
**Pull request by mirkoboehm**
And some related refactoring tasks.

- #293 is a follow-up task.
- #258 is also a follow-up task.
- The command line reference still needs to be integrated into the web site build.

Resolves #286.

### CODE REVIEW

File: lib/go-qmstr/cli-qmstr/run.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+			return fmt.Errorf("no command specified")
+		}
+		return nil
+	},
+	Run: executeRun,
+}
+
+func init() {
+	rootCmd.AddCommand(runCmd)
+	runCmd.Flags().SetInterspersed(false) // this stops flag parsing at the first argument, allowing for e.g. "ls -la" as payload
```
Comment (hemarkus) [inline lib/go-qmstr/cli-qmstr/run.go]: What if we ever need to add a second argument?
What was wrong with proper gnu-style as we used it before and stop parsing with --? 
see https://github.com/QMSTR/qmstr-demo/blob/7383eb658c63f78c33c17d8a15db29e9eb172004/demos/curl/build.sh#L31
https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

File: lib/go-qmstr/cli-qmstr/run.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+func executeRun(cmd *cobra.Command, args []string) {
+	exitCode := Run(args)
+	os.Exit(exitCode)
+}
+
+// Run does everything
+// It also makes sure that even though os.Exit() is called later, all defered functions are properly called.
+func Run(payloadCmd []string) int {
+	// Remind me why Go has no assertions?
```
Comment (hemarkus) [inline lib/go-qmstr/cli-qmstr/run.go]: ??

File: lib/go-qmstr/cli-qmstr/run.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	if len(payloadCmd) == 0 && !keepTmpDirectories {
+		panic(fmt.Errorf("command validation constraint violated"))
```
Comment (hemarkus) [inline lib/go-qmstr/cli-qmstr/run.go]: An error occurred

File: lib/go-qmstr/cli/create.go [hunk 4]
```diff
@@ -16,27 +16,27 @@ var cmdFlags *pflag.FlagSet
 
 var createCmd = &cobra.Command{
 	Use:   "create",
-	Short: "create a new node",
+	Short: "Create a new node",
 	Long:  "create a new node described by a node identifier",
```
Comment (hemarkus) [inline lib/go-qmstr/cli/create.go]: Why does short description start capitalized and long does not?

File: lib/go-qmstr/cli-qmstr/root.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+			Debug = golog.New(os.Stderr, "DEBUG: ", golog.Ldate|golog.Ltime)
+		} else {
+			Debug = golog.New(ioutil.Discard, "", golog.Ldate|golog.Ltime)
+		}
+		Log = golog.New(os.Stderr, "", golog.Ldate|golog.Ltime)
+	},
+}
+
+func init() {
+	cobra.OnInitialize(initLogging)
```
Comment (hemarkus) [inline lib/go-qmstr/cli-qmstr/root.go]: Nice. We should have this in qmstrctl as well.

File: lib/go-qmstr/cli/create.go (hunk 4 above)
Comment (mirkoboehm) [reply lib/go-qmstr/cli/create.go]: For the moment, consistency. All the other descriptions I found are the same way.

File: lib/go-qmstr/cli-qmstr/run.go (hunk 1 above)
Comment (mirkoboehm) [reply lib/go-qmstr/cli-qmstr/run.go]: Discussed and resolved.

**Pull request merged**